data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template

```diff
@@ -152,11 +140,45 @@ $IPTABLES -t raw -A DHCP_IRONIC -j DROP
 {{end}}
 
 # Wait for images to be downloaded/ready
-podman wait -i 1000ms ipa-downloader
 podman wait -i 1000ms coreos-downloader
-while ! curl --fail http://localhost/images/rhcos-ootpa-latest.qcow2.md5sum ; do sleep 1; done
-while ! curl --fail --head http://localhost/images/ironic-python-agent.initramfs ; do sleep 1; done
-while ! curl --fail --head http://localhost/images/ironic-python-agent.kernel ; do sleep 1; done
+
+export KUBECONFIG=/opt/openshift/auth/kubeconfig-loopback
+
+mkdir -p /tmp/nmstate
+
+{{range .PlatformData.BareMetal.Hosts}}
+    until oc get -n openshift-machine-api baremetalhost {{.Name}}; do
+        echo Waiting for Host {{.Name}} to appear...
+        sleep 10
+    done
+    secret_name=$(oc get -n openshift-machine-api baremetalhost {{.Name}} -o jsonpath="{.spec.preprovisioningNetworkDataName}")
+    if [ -n "${secret_name}" ]; then
+        until oc get -n openshift-machine-api secret "${secret_name}"; do
+            echo Waiting for Secret "${secret_name}" to appear...
+            sleep 10
+        done
+        oc get -n openshift-machine-api secret ${secret_name} -o jsonpath="{.data.nmstate}" | base64 -d > /tmp/nmstate/{{.Name}}.yaml
+    else
+        touch /tmp/nmstate/{{.Name}}.yaml
+    fi
+{{end}}
+
+
+# Embed agent ignition into the rhcos live iso
+sudo podman run -d -p 8084:8084 --privileged --name image-customization \
+    --env DEPLOY_ISO="/shared/html/images/ironic-python-agent.iso" \
+    --env DEPLOY_INITRD="/shared/html/images/ironic-python-agent.initramfs" \
+    --env IRONIC_BASE_URL="http://{{.PlatformData.BareMetal.APIVIP}}" \
+    --env IRONIC_RAMDISK_SSH_KEY="$IRONIC_RAMDISK_SSH_KEY" \
+    --env IRONIC_AGENT_IMAGE="$IRONIC_AGENT_IMAGE" \
+    --env IRONIC_AGENT_PULL_SECRET="{{.PlatformData.BareMetal.PullSecretBase64}}" \
+    --env PROVISIONING_INTERFACE=$PROVISIONING_NIC \
+    --env REGISTRIES_CONF_PATH=/tmp/containers/registries.conf \
+    --entrypoint '["/image-customization-server", "--nmstate-dir=/tmp/nmstate/", "--images-publish-addr=http://0.0.0.0:8084"]' \
```

Comment:
Hardcoding 0.0.0.0 may not play well with IPv6-only environments.

Reply:
This is not actually used anywhere - the server doesn't report back the URL to us (it's not like 0.0.0.0 would work either). We just infer what the URL will be. So I think it's fine.

Reply:
Nope, we just need something url-like. We should probably get rid of this parameter and generate the default internally - there are two binaries in the package (this static server and a k8s controller) and a sane URL is needed only for the latter, but the args to both are the same.